pinot-server/src/main/java/org/apache/pinot/server/starter/helix/HelixServerStarter.java

```diff
@@ -103,10 +107,9 @@
  */
 public class HelixServerStarter implements ServiceStartable {
   private static final Logger LOGGER = LoggerFactory.getLogger(HelixServerStarter.class);
-
   private final String _helixClusterName;
   private final String _zkAddress;
-  private final PinotConfiguration _serverConf;
+  private PinotConfiguration _serverConf;
```

Comment:
If we override the configs in the constructor, we can keep this as final.
can we not do that ?
populating the instance config can happen on startup.